File: CRM-FRONT/crm/src/services/partnerService.js
import axiosInstance from "../utils/axiosInstance";

const BASE_URL = "/partners";

const partnerService = {
  // POST /partners/register
  registerPartner: async (partnerData) => {
    const response = await axiosInstance.post(`${BASE_URL}/register`, partnerData);
    return response.data;
  },

  // GET /partners/{partnerId}
  getPartnerById: async (partnerId) => {
    const response = await axiosInstance.get(`${BASE_URL}/${partnerId}`);
    return response.data;
  },

  // PUT /partners/{partnerId}
  updatePartner: async (partnerId, updatedData) => {
    const response = await axiosInstance.put(`${BASE_URL}/${partnerId}`, updatedData);
    return response.data;
  },

  // GET /partners?companyName=...&status=...&page=0&size=10
  searchPartners: async (criteria = {}, page = 0, size = 10) => {
    const params = {
      ...criteria,
      page,
      size
    };
    const response = await axiosInstance.get(BASE_URL, { params });
    return response.data;
  }
};

export default partnerService;

File: partner/src/main/java/com/crm/partner/services/PartnerService.java
package com.crm.partner.services;

import com.crm.partner.entities.Partner;
import com.crm.partner.entities.dto.SearchPartnerCriteria;
import com.crm.partner.entities.dto.request.PartnerRegisterRequestDTO;
import com.crm.partner.entities.dto.request.PartnerUpdateRequestDTO;
import com.crm.partner.entities.dto.response.PartnerProfileResponseDTO;
import com.crm.partner.entities.dto.response.PartnerRegisterResponseDTO;
import com.crm.partner.entities.dto.response.PartnerUpdateResponseDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.Optional;

public interface PartnerService {

    PartnerRegisterResponseDTO registerPartner(PartnerRegisterRequestDTO request) throws Exception;

    PartnerProfileResponseDTO getPartnerById(String partnerId);

    PartnerUpdateResponseDTO updatePartner(String partnerId, PartnerUpdateRequestDTO request);

    Page<PartnerProfileResponseDTO> searchPartners(SearchPartnerCriteria criteria, Pageable pageable);

    PartnerProfileResponseDTO getPartnerByCode(String partnerCode);

}
